tasks/move-granules/index.js

```diff
@@ -234,60 +244,64 @@ async function moveFileRequest(file, sourceBucket, duplicateHandling, buckets) {
 }
 
 /**
-* Move all files in a collection of granules from staging location fo final location,
-* and update granule files to include renamed files if any.
-*
-* @param {Object} granulesObject - an object of the granules where the key is the granuleId
-* @param {string} sourceBucket - source bucket location of files
-* @param {string} duplicateHandling - how to handle duplicate files
-* @param {BucketsConfig} buckets - BucketsConfig instance
-* @returns {Object} the object with updated granules
-**/
-async function moveFilesForAllGranules(granulesObject, sourceBucket, duplicateHandling, buckets) {
+ * Move all files in a collection of granules from staging location fo final location,
+ * and update granule files to include renamed files if any.
+ *
+ * @param {Object} granulesObject - an object of the granules where the key is the granuleId
+ * @param {string} sourceBucket - source bucket location of files
+ * @param {string} duplicateHandling - how to handle duplicate files
+ * @param {BucketsConfig} bucketsConfig - BucketsConfig instance
+ * @returns {Object} the object with updated granules
+ **/
+async function moveFilesForAllGranules(
+  granulesObject,
+  sourceBucket,
+  duplicateHandling,
+  bucketsConfig
+) {
   const moveFileRequests = Object.keys(granulesObject).map(async (granuleKey) => {
     const granule = granulesObject[granuleKey];
+    // TODO [MHS, 2019-02-05] replace with test for cmrfile
     const cmrFileFormat = /.*\.cmr\.xml$/;
     const filesToMove = granule.files.filter((file) => !file.name.match(cmrFileFormat));
     const cmrFiles = granule.files.filter((file) => file.name.match(cmrFileFormat));
     const filesMoved = await Promise.all(
-      filesToMove.map((file) => moveFileRequest(file, sourceBucket, duplicateHandling, buckets))
+      filesToMove.map(
+        (file) => moveFileRequest(file, sourceBucket, duplicateHandling, bucketsConfig)
+      )
+    );
+    const markDuplicates = false;
+    const cmrFilesMoved = await Promise.all(
+      cmrFiles.map(
+        (file) => moveFileRequest(file, sourceBucket, 'replace', bucketsConfig, markDuplicates)
```

Comment:
This is really the big change here. This is effectively what was happening before, but now it's done first and it's explicit.
a cmr xml file is found in the move-granules task.  We move it from the staging location to it's final location, replacing any file that was there and we don't mark it as a duplicate.

I'm not sure this is the desired behavior, but if it's not, we should add a ticket to change it.

Reply:
I think it's fine to replace without marking as dup. Moving granules via the API will reject the move if a file would be overwritten.

The concern I would have here (and this may still be a concern with the way it's done now) is that the CMR file would get moved and the task could fail before the other files were moved which would have files end up in 2 different locations and you'd lose the CMR file.

Reply:
I think this is a problem, not sure how you want to address it, but can it be a different ticket?